JSON tests fail on Go 1.13

Hi @achille-roussel, first I want to say you did an awesome job with this library 💯

According to the [go.mod](https://github.com/segmentio/encoding/blob/master/go.mod#L3) file this package should be compatible with `Go 1.13`, but when I run `go test ./... -v -race` on this version it seems to fail. Should this be correct since we are only testing on `Go 1.14` <https://github.com/segmentio/encoding/blob/master/.circleci/config.yml#L5>?
```go
=== RUN   TestCompact
--- FAIL: TestCompact (0.00s)
##[error]    golang_scanner_test.go:73: Compact(`{"":"<>&  "}`) = `{"":"<>&\u2028\u2029"}`, want original
##[error]    golang_scanner_test.go:81: Compact("{\n\t\"\": \"<>&\u2028\u2029\"\n}") = `{"":"<>&\u2028\u2029"}`, want `{"":"<>&  "}`
=== RUN   TestCompactSeparators
--- FAIL: TestCompactSeparators (0.00s)
##[error]    golang_scanner_test.go:100: Compact("{\"\u2028\": 1}") = "{\"\\u2028\":1}", want "{\"\u2028\":1}"
##[error]    golang_scanner_test.go:100: Compact("{\"\u2029\" :2}") = "{\"\\u2029\":2}", want "{\"\u2029\":2}"

=== RUN   TestGithubIssue11
--- PASS: TestGithubIssue11 (0.00s)
##[error]    json_test.go:1228: json: unsupported value: NaN
=== RUN   TestGithubIssue13
--- PASS: TestGithubIssue13 (0.00s)
##[error]    json_test.go:1249: {"Stringer":null,"MyInt":0}

--- PASS: TestDecodeLines (0.00s)
    --- PASS: TestDecodeLines/bare_object (0.00s)
    --- PASS: TestDecodeLines/multiple_objects_on_one_line (0.00s)
    --- PASS: TestDecodeLines/object_spanning_multiple_lines (0.00s)
    --- PASS: TestDecodeLines/trailing_newline (0.00s)
    --- PASS: TestDecodeLines/multiple_trailing_newlines (0.00s)
    --- PASS: TestDecodeLines/blank_lines (0.00s)
    --- PASS: TestDecodeLines/no_trailing_newline (0.00s)
    --- PASS: TestDecodeLines/leading_whitespace (0.00s)
    --- PASS: TestDecodeLines/one_object,_multiple_reads (0.00s)
    --- PASS: TestDecodeLines/one_object_+_EOF (0.00s)
    --- PASS: TestDecodeLines/leading_whitespace_+_EOF (0.00s)
    --- PASS: TestDecodeLines/multiple_objects_+_EOF (0.00s)
    --- PASS: TestDecodeLines/one_object_+_multiple_reads_+_EOF (0.00s)
    --- PASS: TestDecodeLines/multiple_objects_+_multiple_reads_+_EOF (0.00s)
    --- PASS: TestDecodeLines/unmarshal_error_while_decoding (0.00s)
##[error]        json_test.go:735: unmarshal error json: cannot unmarshal "42}" into Go struct field json.obj.Good. of type bool
    --- PASS: TestDecodeLines/unmarshal_error_while_decoding_last_object (0.00s)
##[error]        json_test.go:735: unmarshal error json: cannot unmarshal "42}" into Go struct field json.obj.Good. of type bool

--- a/README.md
+++ b/README.md
@@ -18,6 +18,21 @@ performance library, we also aim for very low development and maintenance overhe
 by implementing APIs that can be used as drop-in replacements for the default
 solutions.
 
+## Requirements and Maintenance Schedule
+
+This package has no dependencies outside of the core runtime of Go.  It
+requires a recent version of Go.
+
+This package follows the same maintenance schedule as the [Go
+project](https://github.com/golang/go/wiki/Go-Release-Cycle#release-maintenance),
+meaning that issues relating to versions of Go which aren't supported by the Go
+team, or versions of this package which are older than 1 year, are unlikely to
+be considered.
+
+Additionally, we have fuzz tests which aren't a runtime required dependency but
+will be pulled in when running `go mod tidy`.  Please don't include these go.mod
+updates in change requests.
+
 ## encoding/json [![GoDoc](https://godoc.org/github.com/segmentio/encoding/json?status.svg)](https://godoc.org/github.com/segmentio/encoding/json)
 
 More details about the implementation of this package can be found [here](json/README.md).

--- a/go.mod
+++ b/go.mod
@@ -1,3 +1,3 @@
 module github.com/segmentio/encoding
 
-go 1.13
+go 1.14

